### DISCUSSION TITLE
Enable "production-level" on-screen logging, versus showing stack traces (aka "error boundary")

### DISCUSSION THREAD

**Discussion by bandaider** [h2oai/wave#2192] [Q&A]
#### Is your feature request related to a problem?

There appears to be no way to display a user-friendly error message when one's application fails due to developer or runtime error.

For example, in React, this is considered an "error boundary". When an unhandled exception occurs in your application, it is captured in the error boundary, where the developer can provide a helpful message to the user.

Currently, there appears no way to specify or add such a boundary, at least out of the box.

#### Describe the solution you'd like

When the application is run in a "production" mode, perhaps through a new flag, the developer can define or set a boundary area or qargs that is captured whereby the developer can render a card that hides the underlying exception but allows the developer to specify a user-friendly error message.

#### Describe alternatives you've considered

None at this time.

#### Additional context

None at this time.

**Comment by mturoci** [comment]
Hi @bandaider. Did you try using Python's `try catch`? Wave only shows stacktrace for unhandled errors. However, app developers are free to handle and display the errors in any way they want.

[Here](https://github.com/h2oai/h2o-llmstudio/blob/main/llm_studio/app_utils/handlers.py#L359-L364) is one possible implementation that one of our teams uses in an app called LLM Studio.
